lib/spack/docs/basic_usage.rst

```diff
@@ -698,8 +699,10 @@ without the need to set ``LD_LIBRARY_PATH``.
           f77: /opt/gcc/bin/gfortran
           fc: /opt/gcc/bin/gfortran
         environment:
-          set:
-            LD_LIBRARY_PATH : /opt/gcc/lib
+          - [unset, BAD_VARIABLE]
```

Comment:
Because the order of modifications is unclear if we use a dictionary (even if we use an ordered one, users still have to make sure that it's ordered). What will be the value of `MY_VAR` if we have the following?
```yaml
set:
  MY_VAR: /ololo
unset:
  MY_VAR
prepend-path:
  MY_VAR: /alala
append-path:
  MY_VAR: /ilili
```
Yes, we could just set `MY_VAL` to the value we need, but would we need to allow for such confusing constructs?

Reply:
Ordering is good, but I don't like using a list for something that is semantically a map. Since we are essentially associating values with keys, I'd prefer an ordered dict to a list.

`[unset, MY_VAR]` feels convoluted compared to `unset: MY_VAR`.